packages/firestore/test/unit/specs/spec_test_runner.ts

```diff
@@ -984,10 +984,14 @@ abstract class TestRunner {
     this.validateLimboDocs();
     // Always validate that the expected active targets match the actual active
     // targets
-    this.validateActiveTargets();
+    await this.validateActiveTargets();
   }
 
   private validateLimboDocs(): void {
+    if (!this.started) {
```

Comment:
uh... this looks a bit odd and should perhaps be commented (why do we call validateLimboDocs() when we're not started?)

Reply:
I moved this up to `validateStateExpectations` and added a comment. It's a bit clearer now - but I am now also not validating the active targets (which we might still want to do).

Another option would be to reset the limbo docs to an empty array in SyncEngine and call `.expectLimboDocs()` after `.shutdown` in the spec tests.